lib/src/components/NxFilterInput/__tests__/NxFilterInput.test.tsx

```diff
@@ -5,160 +5,109 @@
-        expect(icon).toHaveClassName('nx-icon--filter-icons');
-      }
-  );
-
-  it('passes an faSearch icon with nx-icon--filter-icons as the prefixContent when searchIcon is true', function() {
-    const IconFixture = function() {
-          return shallowComponent({ searchIcon: true }).prop('prefixContent');
-        },
-        icon = shallow(<IconFixture />);
-
-    expect(icon).toMatchSelector(NxFontAwesomeIcon);
-    expect(icon).toHaveProp(icon, faSearch);
-    expect(icon).toHaveClassName('nx-icon--filter-icons');
+  it('renders a filter icon when searchIcon is undefined', function() {
+    const icons = quickRender().getAllByRole('img', { hidden: true });
+
+    expect(icons[0]).toHaveAttribute('data-icon', 'filter');
   });
 
-  it('passes an NxButton with the Close icon and nx-btn--clear classname as the suffixContent', function() {
-    const BtnFixture = function() {
-          return shallowComponent().prop('suffixContent');
-        },
-        btn = shallow(<BtnFixture />);
+  it('renders a search icon when searchIcon is true', function() {
+    const icons = quickRender({ searchIcon: true }).getAllByRole('img', { hidden: true });
 
-    expect(btn).toExist();
-    expect(btn).toMatchSelector(NxButton);
-    expect(btn.children()).toMatchSelector(Close);
-    expect(btn).toHaveClassName('nx-btn--clear');
+    expect(icons[0]).toHaveAttribute('data-icon', 'search');
   });
 
-  it('sets the title as "Clear filter" to the Nxbutton when searchIcon is undefined', function() {
-    const BtnFixture = function() {
-          return shallowComponent().prop('suffixContent');
-        },
-        btn = shallow(<BtnFixture />);
+  it('renders a button with an accessible name of "Clear filter" when searchIcon is undefined', async function() {
+    const clearBtn = quickRender().getByRole('button');
 
-    expect(btn).toHaveProp('title', 'Clear filter');
+    expect(clearBtn).toBeInTheDocument();
+    await waitFor(() => expect(clearBtn).toHaveAccessibleName('Clear filter'));
   });
 
-  it('sets the title as "Clear search" to the Nxbutton when searchIcon is true', function() {
-    const BtnFixture = function() {
-          return shallowComponent({ searchIcon: true }).prop('suffixContent');
-        },
-        btn = shallow(<BtnFixture />);
+  it('renders a button with an accessible name of "Clear search" when searchIcon is true', async function() {
+    const clearBtn = quickRender({ searchIcon: true }).getByRole('button');
 
-    expect(btn).toHaveProp('title', 'Clear search');
+    expect(clearBtn).toBeInTheDocument();
+    await waitFor(() => expect(clearBtn).toHaveAccessibleName('Clear search'));
   });
 
-  it('fires onChange with the empty string when the Escape key is pressed', function() {
-    const onKeyDown = jest.fn(),
+  it('fires onChange with the empty string when the Escape key is pressed', async function() {
+    const user = userEvent.setup(),
+        onKeyDown = jest.fn(),
```

Comment:
Do you need `onKeyDown` here?

Reply:
No, thanks for catching it, updated here c801738d2fa4bbfeb10c78ace8bf91b46a0ee65b.